ios/accessibility/accessibility_control.go

```diff
@@ -52,32 +69,67 @@ type AXElementData struct {
 	SpokenDescription    string `json:"spokenDescription"`    // Spoken description of the element
 }
 
-func (a ControlInterface) readhostAppStateChanged() {
+func (a *ControlInterface) readhostAppStateChanged() {
 	for {
 		msg := a.channel.ReceiveMethodCall("hostAppStateChanged:")
 		stateChange, err := nskeyedarchiver.Unarchive(msg.Auxiliary.GetArguments()[0].([]byte))
 		if err != nil {
-			panic(err)
+			log.Errorf("Error unarchiving app state change: %v", err)
+			continue
 		}
 		value := stateChange[0]
 		log.Infof("hostAppStateChanged:%s", value)
 	}
 }
 
-func (a ControlInterface) readhostInspectorNotificationReceived() {
+// Subscribe returns a read-only channel for the consumer and a "cancel" function to unsubscribe
+func (a *ControlInterface) Subscribe() (<-chan Notification, func()) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	// create a new buffered channel for this specific consumer
+	ch := make(chan Notification, 100)
```

Comment:
I guess the idea here is to prevent anything from blocking, but what happens after 100 notifications? we just delay blocking in that case. I would rather use unbuffered channels so that that issue can be spotted immediately and not with some delay